From 25f394826725e4e6039d30a89f5d8cb5e36fb492 Mon Sep 17 00:00:00 2001
From: Jason Chen <b02280@freescale.com>
Date: Mon, 20 Dec 2010 14:28:22 +0800
Subject: [PATCH 133/280] ENGR00137167 ipu lib: make fb yoffset equal 0 during
 init

As kernel will keep fb yoffset, we need set yoffset to 0 for keeping
our ipu lib usage.

Signed-off-by: Jason Chen <b02280@freescale.com>
---
 ipu/mxc_ipu_hl_lib.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ipu/mxc_ipu_hl_lib.c b/ipu/mxc_ipu_hl_lib.c
index d97164c..64d83e9 100644
--- a/ipu/mxc_ipu_hl_lib.c
+++ b/ipu/mxc_ipu_hl_lib.c
@@ -970,6 +970,8 @@ done:
 static int fit_fb_setting(struct fb_var_screeninfo * var, int width,
 	int height, int fmt, ipu_channel_t fb_chan, int bufs)
 {
+	if (var->yoffset != 0)
+		return 0;
 	if (fb_chan == MEM_BG_SYNC)
 		return ((var->xres_virtual == var->xres) &&
 			(var->yres_virtual == bufs*var->yres));
@@ -1288,6 +1290,8 @@ again:
 				fb_var.yres_virtual = fb_var.yres * fbbufs;
 			}
 
+			fb_var.yoffset = 0;
+
 			if ( ioctl(ipu_priv_handle->output.fd_fb, FBIOPUT_VSCREENINFO, &fb_var) < 0) {
 				dbg(DBG_ERR, "Set FB var info failed!\n");
 				close(ipu_priv_handle->output.fd_fb);
-- 
1.8.0

